### PR TITLE
#6 Module for fund and organization

### LINKED ISSUE #6: Create module for fund and organization
ℹ️ This issue is part of Epic #2 

## Objective

Create a module to fund used in the smart contract

## Checklist

- [x] Create a module for fund
- [x] Create a module for organization

### PR DESCRIPTION
**Pull request by Jenietoc**
## Description

In this Pr we created the fund and organization module and added a function to the token_operation module.

Fixes #6 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by Odraxs** [comment]
@Jenietoc Let's wait util #13 is merged in `v0.1` then rebase this one, so we can check if  `cargo clippy —all-targets —all-features — -D warnings` return more warning and correct them in this `pr`

File: mtk-automation/src/lib.rs [hunk 1]
```diff
@@ -22,29 +24,44 @@ pub trait OrganizationContractTrait {
     /// revoke to the organization
     fn revoke_m(env: Env, from: AccountId);
 
-    /// compensate a member to the organization
-    fn comp_m(e: Env, token_approval_sig: Signature, to: AccountId, r_type: Symbol);
+    /// reward a member to the organization
```
Comment (einerzg) [inline mtk-automation/src/lib.rs:27]: better `rewarding a member of the organization` or `reward a member of the organization`

File: mtk-automation/src/lib.rs [hunk 2]
```diff
@@ -22,29 +24,44 @@ pub trait OrganizationContractTrait {
...
+    fn reward_m(e: Env, token_approval_sig: Signature, to: AccountId, r_type: Symbol);
 
     /// get token contract to the organization
     fn get_tc_id(env: Env) -> BytesN<32>;
+
+    /// get balanca for the organization
```
Comment (einerzg) [inline mtk-automation/src/lib.rs:33]: I think what they mean here is `get the organization balance`

File: mtk-automation/src/lib.rs [hunk 3]
```diff
@@ -22,29 +24,44 @@ pub trait OrganizationContractTrait {
...
+    fn get_bal(env: Env) -> i128;
     
     /// get members to the organization
     fn get_m(env: Env) -> Vec<AccountId>;
+
+    /// name of organization
```
Comment (einerzg) [inline mtk-automation/src/lib.rs:39]: `get the organization's name`

File: mtk-automation/src/lib.rs [hunk 4]
```diff
@@ -22,29 +24,44 @@ pub trait OrganizationContractTrait {
...
+    fn org_name(env: Env) -> Symbol;
+    
+    /// fund contract balance for organization
```
Comment (einerzg) [inline mtk-automation/src/lib.rs:42]: `fund contract balance for the organization`

**Review by einerzg — CHANGES_REQUESTED** [review]

**Comment by Jenietoc** [comment]
> @Jenietoc Let's wait util #13 is merged into `v0.1` then rebase this one, so we can check if `cargo clippy —all-targets —all-features — -D warnings` returns more warnings and correct them in this `pr`

Ok, I'm waiting for the rebase

**Review by einerzg — APPROVED** [review]

File: mtk-automation/src/contract_actions/fund.rs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+//! Module for Organization
+//! 
+//! Module for managment the fund of the organization
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/fund.rs:3]: Please fix the typo errors and try to write this on a better way

File: mtk-automation/src/contract_actions/organization.rs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+//! Module for Organization
+//! 
+//! Module to set and get organizaion name
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/organization.rs:3]: Fix type errors here too

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]
Everything makes sense, please fix the issues where I left comments

**Review by Odraxs — DISMISSED** [review]

**Review by L-Zuluaga — APPROVED** [review]

**Pull request merged**